src/Microsoft.DotNet.Cli.Utils/CommandResolution/WindowsExePreferredCommandSpecFactory.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.Extensions.PlatformAbstractions;
+using NuGet.Frameworks;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    public class WindowsExePreferredCommandSpecFactory : IPlatformCommandSpecFactory
+    {
+        public CommandSpec CreateCommandSpec(
+           string commandName,
+           IEnumerable<string> args,
+           string commandPath,
+           CommandResolutionStrategy resolutionStrategy,
```

Comment:
Why is the strategy an argument coming into the resolver? It seems this represents the strategy of the successful resolver...